sdk/storage/azure-storage-blob/azure/storage/blob/_shared/models.py

```diff
@@ -351,12 +320,6 @@ def __init__(self, read=False, write=False, delete=False, list=False,  # pylint:
         self.update = update or ('u' in _str)
         self.process = process or ('p' in _str)
```

Comment:
We should construct `self._str` in here, and keep it around (or use directly the input from `from_string`. Then we can return it in `__str__`. This should keep the "non-lossy" behaviour discussed in the review